src/chainstate/coordinator/mod.rs

```diff
@@ -0,0 +1,307 @@
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::{Arc, Mutex};
+use std::sync::mpsc::{self, Sender};
+use std::process;
+
+use burnchains::BurnchainHeaderHash;
+use chainstate::burn::{BlockHeaderHash, BlockSnapshot};
+use chainstate::stacks::{StacksBlock, TransactionPayload};
+use chainstate::stacks::db::StacksHeaderInfo;
+use chainstate::stacks::events::{StacksTransactionReceipt};
+
+#[derive(Debug, Clone)]
+struct PoxId;
+
+#[derive(Debug, Clone)]
+pub struct SortitionId (
+    BurnchainHeaderHash,
+    PoxId
+);
+
+#[derive(Debug, Clone)]
+pub struct StacksBlockId {
+    burnchain_header_hash: BurnchainHeaderHash,
+    header_hash: BlockHeaderHash,
+    pox_identifier: PoxId,
+}
+
+struct ChainStateDB;
+impl ChainStateDB {
+
+    pub fn stubbed() -> ChainStateDB {
+        ChainStateDB {}
+    }
+
+    #[allow(unused_variables)]
+    pub fn is_block_processed(&self, block_id: &StacksBlockId) -> Result<bool, ()> {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn process_blocks(&mut self, burnchain_blocks_db: &mut BurnchainBlocksDB, pox_identifier: PoxId) -> Result<Vec<(Option<(StacksHeaderInfo, Vec<StacksTransactionReceipt>)>, Option<TransactionPayload>)>, ()> {
+        unimplemented!()
+    }
+}
+
+struct BurnchainBlocksDB;
+impl BurnchainBlocksDB {
+
+    pub fn stubbed() -> BurnchainBlocksDB {
+        BurnchainBlocksDB {}
+    }
+
+    pub fn get_canonical_chain_tip(&self) -> BlockSnapshot {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn get_burnchain_block(&self, burnchain_header_hash: &BurnchainHeaderHash) -> Result<BlockSnapshot, ()> {
+        unimplemented!()
+    }
+}
+
+struct BlocksDB;
+impl BlocksDB {
+
+    pub fn stubbed() -> BlocksDB {
+        BlocksDB {}
+    }
+
+    #[allow(unused_variables)]
+    pub fn get_blocks_ready_to_process(&self, sortition_id: &SortitionId, sortition_db: &SortitionDB) -> Option<Vec<StacksBlock>> {
+        // This method will be calling sortition_db::latest_stacks_blocks_processed
+        unimplemented!()
+    }
+}
+
+struct PoxDB;
+impl PoxDB {
+
+    pub fn stubbed() -> PoxDB {
+        PoxDB {}
+    }
+
+    #[allow(unused_variables)]
+    pub fn get_canonical_pox_id(&self, burnchain_header_hash: &BurnchainHeaderHash) -> PoxId {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn get_ordered_missing_anchors(&self, upper_bound: &StacksBlockId) -> Vec<StacksBlockId> {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    // Note: we'd be temporary using an associated function instead of method, because this call is writing. 
+    pub fn process_anchor(block: &StacksBlockId, chain_state: &ChainStateDB) -> Result<(), ()> {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn get_reward_set_start_for(&self, block: &StacksBlockId) -> &BurnchainHeaderHash {
+        unimplemented!()
+    }
+}
+
+struct SortitionDB;
```

Comment:
Same idea here -- I think `SortitionDB` should be a trait, since then we can proceed to implement it for the existing sortition DB.

Reply:
For these and the other notes — I’ve already started the work on “unstubbing” the SortitionDB and other elements here as part of integrating the ChainsCoordinator into the rest of the codebase. I don’t think we should necessarily require that as part of this PR.

Reply:
I'm not sure how to read this code, then.  The names for various structs here are already in use in other modules.  Are we intending to use these struct definitions over the others?